QuerySolver.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+public class QuerySolver
```

Comment:
I would rename it to `DataDownloader` or something. `byte[]` together with `query` got me confused for a while